buglist.cgi

```diff
@@ -656,32 +656,25 @@ if (!$order || $order =~ /^reuse/i) {
 
 my @order_columns;
 if ($order) {
+#
+# 	Convert the value of the "order" form field (mostly affects Advanced Search) into a list of columns by which to sort the results.
+# 	Also as preparation for later calling in "Bugzilla/Search.pm": which expects "order" as array.
+#
+	@order_columns = (
+	$order =~ /^(?:[Bb]ug[ _]?(?:[Nn]umber|ID|id)|b?id|B?ID)$/ ?
+	"bug_id" :
 
-  # Convert the value of the "order" form field into a list of columns
-  # by which to sort the results.
-ORDER: for ($order) {
-    /^Bug Number$/ && do {
-      @order_columns = ("bug_id");
-      last ORDER;
-    };
-    /^Importance$/ && do {
-      @order_columns = ("priority", "bug_severity");
-      last ORDER;
-    };
-    /^Assignee$/ && do {
-      @order_columns = ("assigned_to", "bug_status", "priority", "bug_id");
-      last ORDER;
-    };
-    /^Last Changed$/ && do {
-      @order_columns
-        = ("changeddate", "bug_status", "priority", "assigned_to", "bug_id");
-      last ORDER;
-    };
-    do {
-      # A custom list of columns. Bugzilla::Search will validate items.
-      @order_columns = split(/\s*,\s*/, $order);
-    };
-  }
+	$order =~ /^(?:[Ii]mportance|(?:[Pp]rio|[Ss]eve)rity)$/ ?
+	( "priority", "bug_severity" ) :
+
+	$order =~ /^[Aa]ssign(?:e[ed])?$/ ?
+	( "assigned_to", "bug_status", "priority", "bug_id" ) :
+
+	$order =~ /^(?:(?:[Ll]ast[ _]?)?[Cc]hanged?(?:[ _]?[Dd]ate)?|lc)$/ ?
+	( "changeddate", "bug_status", "priority", "assigned_to", "bug_id" ) :
+
```

Comment:
Where are these modifications coming from? For instance, "Last Changed" becomes /^(?:(?:[Ll]ast[ _]?)?[Cc]hanged?(?:[ _]?[Dd]ate)?|lc)$/ with no reason.

This makes the code unreadable, prone to errors, and I see no reason to change the list of accepted input.

Reply:
    "to allow plausible variations", for extensibility.
    So "order" may be specified as other variants: e.g. "order=lc" in request URL.

    Readability-wise, to represent a string group (and related operation):
    Such RegEx is already the optimal form: no other possibilities possible.

Reply:
That is not true. //i would be much more readable than [Ll], [Cc] and co. Also, I see no valid reason to accept several different inputs for the same thing. In all cases, this should be discussed in a bugzilla bug first.

Reply:
    Is "lAsT CHaNgED" expected..?
    Writing "order=Last%20Changed" in URL is a pain...

Reply:
Expected ? No. But being case-insensitive could make sense.

Reply:
    Of course.
    [ Quote MasterInQuestion @ CE 2023-12-01 04:44:44 UTC:
[https://github<i>.</i>com/curl/curl/discussions/12397#discussioncomment-7674753](https://github.com/curl/curl/discussions/12397#discussioncomment-7674753)
    CLI comparing typical programming languages: ease of input would be emphasized in addition.
    That's why PowerShell style &#x60;`Get-Help Out-String -Full`&#x60; needs shortcut. ]

    See also: [https://github<i>.</i>com/MasterInQuestion/talk/discussions/15#discussion-6022492](https://github.com/MasterInQuestion/talk/discussions/15#discussion-6022492)
    "Interactive vs. Canonical"